keras/data/dataset.py

```diff
@@ -0,0 +1,11 @@
+# TODO add some default like hdf5 dataset, directory, etc
+class Dataset():
+    """
+    Base object for every dataset
```

Comment:
Standardize docstring format and add a more complete description of what datasets are